src/commands/oas2json.js

```diff
@@ -1,40 +1,59 @@
 import { Command } from 'commander'
+import filenamify from 'filenamify'
 import fs from 'fs-extra'
-import YAML from 'yaml'
+import _get from 'lodash.get'
+import _trimStart from 'lodash.trimstart'
 import path from 'path'
-import { exit } from 'process'
 import pino from 'pino'
+import { exit } from 'process'
+import YAML from 'yaml'
+
 import { fromSchema } from '../utils/openapi-schema-to-json-schema-wrapper.cjs'
 
 const COMPONENT_REF_REGEXP = /#\/components\/schemas\/[^"]+/g
 
-export const adaptSchema = (generatedSchema, name) => {
+export const adaptSchema = (generatedSchema, name, filename) => {
   delete generatedSchema.$schema
   generatedSchema.title = name
-  generatedSchema.$id = `${name}.json`
+  generatedSchema.$id = `${filename}.json`
 
   if (generatedSchema.format?.includes('date')) {
     generatedSchema.tsType = 'Date'
   }
 }
 
-const processSchema = (name, schema, schemasPath) => {
-  const generatedSchema = fromSchema(schema)
-  adaptSchema(generatedSchema, name)
+const processSchema = (name, schema, schemasPath, logger, dirname = '') => {
+  let generatedSchema
+  try {
+    generatedSchema = fromSchema(schema)
+  } catch (error) {
+    logger.warn('Failed to convert non-object attribute, skipping')
+    return
+  }
+
+  const filename = _trimStart(filenamify(name, { replacement: '-' }), '-')
+  adaptSchema(generatedSchema, name, filename)
 
   let stringifiedSchema = JSON.stringify(generatedSchema, undefined, 2)
-  const results = stringifiedSchema.match(COMPONENT_REF_REGEXP)
+  const schemaRefs = stringifiedSchema.match(COMPONENT_REF_REGEXP)
 
-  ;(results || []).forEach(element => {
-    const refName = element.split('/').slice(-1)
+  ;(schemaRefs || []).forEach(element => {
+    let refName = element.split('/').slice(-1)
     stringifiedSchema = stringifiedSchema.replace(element, `${refName}.json`)
```

Comment:
Just a note: the library supports now any component definition path BUT these lines are still only replacing ref values when they look like `#/components/schemas/Foo`. Not sure whether we should take action now, but maybe let's drop a comment to point out this limitation.